### PR TITLE
deps module utils: add function failed() to retrieve dependencies check result

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Add function `failed()` to module `deps`, providing the ability to check the dependency check result without triggering an exception.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/deps.py

### CODE REVIEW

**Comment by russoz** [comment]
The CI is failing because of `iso_customize`. It looks like the error is not related to change performed in this PR, and other modules using `deps` were unaffected by the change.

I suppose that module was not yet tested in those configurations (RHEL 7.9, RHEL 9.1, MacOS 13.2). @ZouYuhua could you please test your module independently. It looks like something changed in the execution environment that affected your module. Maybe Python 3.11, but I am not sure.

**Comment by felixfontein** [comment]
It's likely also failing in the nightlies (they fail for some days, but I didn't got around to check why exactly). It's potentially a newer version of some library...

File: plugins/module_utils/deps.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def message(self):
     def failed(self):
         return self.state == 1
 
-    def verify(self, module):
+    def validate(self, module):
```
Comment (felixfontein) [inline plugins/module_utils/deps.py:53]: This is a breaking change. You should add a wrapper with the old name that calls the new one, and also `module.deprecate()` to announce that some code needs to be changed :)

**Comment by felixfontein** [comment]
(And this needs a changelog fragment.)

File: plugins/module_utils/deps.py (hunk 1 above)
Comment (russoz) [reply plugins/module_utils/deps.py:53]: This is inside the classe `_Dependency`, no one should be using that. No object of this class is returned to code outside this file.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
Failing tests are unrelated.

**Comment by felixfontein** [comment]
@russoz thanks!

**Pull request merged**